metricflow/plan_conversion/dataflow_to_sql.py

```diff
@@ -877,12 +901,18 @@ def visit_combine_metrics_node(self, node: CombineMetricsNode) -> SqlDataSet:
 
         parent_data_sets: List[AnnotatedSqlDataSet] = []
         table_alias_to_metric_instances: OrderedDict[str, Tuple[MetricInstance, ...]] = OrderedDict()
+        table_alias_to_measure_instances: OrderedDict[str, Tuple[MeasureInstance, ...]] = OrderedDict()
 
         for parent_node in node.parent_nodes:
             parent_sql_data_set = parent_node.accept(self)
             table_alias = self._next_unique_table_alias()
             parent_data_sets.append(AnnotatedSqlDataSet(data_set=parent_sql_data_set, alias=table_alias))
-            table_alias_to_metric_instances[table_alias] = parent_sql_data_set.instance_set.metric_instances
+            if parent_sql_data_set.instance_set.metric_instances:
+                table_alias_to_metric_instances[table_alias] = parent_sql_data_set.instance_set.metric_instances
+            elif parent_sql_data_set.instance_set.measure_instances:
+                table_alias_to_measure_instances[table_alias] = parent_sql_data_set.instance_set.measure_instances
+            else:
+                raise RuntimeError("Attempting to combine output nodes without any measures/metrics upstream")
```

Comment:
If we use a shared protocol we don't need to do this, although it might be cleaner if we update the type on table_alias_to_metric_instances to take the specs instead. We only use the instance to get the spec in the one callsite here.

Reply:
this was recently changed to use instance due to the fill null https://github.com/dbt-labs/metricflow/pull/850/files#diff-488be53d5a33d76e334c75668b62f8c961f169996b1fba1cca5daaf177e1d81bR900